Source/WebCore/xml/XPathFunctions.cpp

```diff
@@ -434,18 +434,30 @@ Value FunConcat::evaluate() const
     result.reserveCapacity(1024);
 
     for (unsigned i = 0, count = argumentCount(); i < count; ++i) {
+        auto& evaluationContext = Expression::evaluationContext();
+        auto backupContext = evaluationContext;
+
         String str(argument(i).evaluate().toString());
         result.append(str);
+
+        evaluationContext = backupContext;
     }
 
     return result.toString();
 }
 
 Value FunStartsWith::evaluate() const
 {
+    auto& evaluationContext = Expression::evaluationContext();
+    auto backupContext = evaluationContext;
+
     String s1 = argument(0).evaluate().toString();
+
+    std::swap(evaluationContext, backupContext);
```

Comment:
Why is swap correct here rather than assignment?

Reply:
This is about what we want Expression::evaluationContext() to be at the end of this function.
Here, at the end of the function, we want Expression::evaluationContext() to be the context after evaluating the first argument, not the context before evaluating the first argument. As a result, we have to back up the context right after evaluating the first argument.